docs/src/developing/programming-model/transactions.md

```diff
@@ -171,6 +171,26 @@ that this method only supports fixed sized types. Token utilizes the
 trait to encode/decode instruction data for both token instructions as well as
 token account states.
 
+### Multiple instructions in a single transaction
+
+A transaction can contain instructions in any order.   This means a malicious
+user could craft transactions that may pose instructions in an order that the
+program has not been protected against.  Programs should be hardened to properly
+and safely handle any possible instruction sequence.
+
+One not so obvious example is account deinitialization.  Some programs may
+attempt to deinitialize an account by setting its lamports to zero, with the
+assumption that the runtime will delete the account.  This assumption may be
+valid between transactions, but it is not between instructions or cross-program
+invocations.  To harden against this, the program should also explicitly zero out the
+account's data.
+
+An example of where this could be a problem is if a token program, upon
+transferring the token out of an account, sets the account's lamports to zero,
+assuming it will be deleted by the runtime.  If the program does zero out the
```

Comment:
Maybe, `If the program does zero out` should be `If the program doesn't zero out`?

I think you just forgot to add `not` because a malicious user *couldn't* trail if the account has been zeroed-out.

Reply:
@jackcmay it looks like @jstarry (thanks!) fixed this for us..: https://github.com/solana-labs/solana/commit/3eea88a3a84b278efc8d980f60c3b0d69dad0cbb

```patch
diff --git a/docs/src/developing/programming-model/transactions.md b/docs/src/developing/programming-model/transactions.md
index ce376ddaa5f..6fe4093360f 100644
--- a/docs/src/developing/programming-model/transactions.md
+++ b/docs/src/developing/programming-model/transactions.md
@@ -187,7 +187,7 @@ account's data.
 
 An example of where this could be a problem is if a token program, upon
 transferring the token out of an account, sets the account's lamports to zero,
-assuming it will be deleted by the runtime.  If the program does zero out the
+assuming it will be deleted by the runtime.  If the program does not zero out the
 account's data, a malicious user could trail this instruction with another that
 transfers the tokens a second time.
```

Could you quickly check this?. Also, how about backporting to v1.5 this as well?

Reply:
@ryoqun @jstarry pushed to v1.5: https://github.com/solana-labs/solana/commit/eacf9209f778dc0d7844f1e726be47a23651b4f0

Reply:
Ah sorry for forgetting the backport!